libraries/AP_Common/Location.cpp

```diff
@@ -110,17 +110,23 @@ bool Location::get_alt_cm(AltFrame desired_frame, int32_t &ret_alt_cm) const
         AP_HAL::panic("Should not be called on invalid location: Location cannot be (0, 0, 0)");
     }
 #endif
-    Location::AltFrame frame = get_alt_frame();
+    Location::AltFrame current_frame = get_alt_frame();
 
+    return get_spec_alt_cm(current_frame, desired_frame, alt, ret_alt_cm);
+}
+
+// mday99: This function was a part of get_alt_cm up above. I pulled it out to accept any altitude as an input
+bool Location::get_spec_alt_cm(AltFrame current_frame, AltFrame desired_frame, int32_t alt_conv, int32_t &ret_alt_cm) const
```

Comment:
```suggestion
bool Location::get_spec_alt_cm(AltFrame frame, AltFrame desired_frame, int32_t alt, int32_t &ret_alt_cm) const
```

`static` means you don't need to avoid the name conflicts.